### PR TITLE
feat(lint): add support for GitHub Actions output format

### PR DESCRIPTION
**Pull request by rjelierse**
## What/Why/How?

Add a new output format for the `lint` subcommand that can be used in a GitHub Actions workflow directly.

## Check yourself

- [x] Code changed? - Tested with redoc/reference-docs/workflows (internal)
- [x] All new/updated code is covered with tests
- [x] New package installed? - Tested in different environments (browser/node)

## Security

- [ ] Security impact of change has been considered
- [ ] Code follows company security practices and guidelines

### CODE REVIEW

File: docs/commands/lint.md [hunk 1]
```diff
@@ -228,6 +228,17 @@ Due to the limitations of this format, only file name, line, column, severity,
 and rule ID (in the `source` attribute) are included. All other information is
 omitted.
 
+#### GitHub Actions
+
+The `lint` command also comes with support for a [GitHub Actions](https://docs.github.com/en/actions) specific formatting.
+Use `redocly lint --format=github-actions` to have any encountered problem annotated on the affected files.
```
Comment (adamaltman) [inline docs/commands/lint.md:234]: 🔥 🔥 🔥

File: .changeset/tame-gifts-tease.md [hunk 2]
```diff
@@ -0,0 +1,6 @@
+---
+"@redocly/openapi-core": patch
+"@redocly/cli": patch
```
Comment (adamaltman) [inline .changeset/tame-gifts-tease.md:3]: Would we classify this as a patch or feature?

It seems like a new feature to me and would result in a new minor version?

I didn't review the substance of the PR but the spirit is awesome. 🏆 

Thank you @rjelierse

**Review by RomanHotsiy — APPROVED** [review]
This is great! Thanks @rjelierse 🔥 

(approving it but please adjust the changeset from patch to minor, or we can do it too).

**Comment by tatomyr** [comment]
@rjelierse I wonder why you've changed the package-lock file. Is that intentional?

File: .changeset/tame-gifts-tease.md (hunk 2 above)
Comment (rjelierse) [reply .changeset/tame-gifts-tease.md:3]: I was debating whether this should be considered a new function, but new minor function works for me.

**Comment by rjelierse** [comment]
> @rjelierse I wonder why you've changed the package-lock file. Is that intentional?

It's a leftover from testing with local version of `@redocly/openapi-core`, I will undo the change to `package-lock.json`

**Review by lornajane — APPROVED** [review]
I really like this! The change looks good and the docs are great too, thanks @rjelierse!

File: .changeset/tame-gifts-tease.md [hunk 3]
```diff
@@ -0,0 +1,6 @@
+---
+"@redocly/openapi-core": minor
+"@redocly/cli": minor
+---
+
+Add support for `github-actions` as an output format for the `redocly lint` command. This will annotate reported problems on files when used in a workflow on GitHub Actions.
```
Comment (lornajane) [inline .changeset/tame-gifts-tease.md:6]:
```suggestion
Added support for a `github-actions` output format for the `lint` command to annotate reported problems on files when used in a GitHub Actions workflow.
```

**Review by adamaltman — APPROVED** [review]

**Pull request merged**